contracts/lib/Select.sol

```diff
@@ -0,0 +1,93 @@
+pragma solidity ^0.4.24 ;
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+library Select {
+    using SafeMath for uint256;
+
+    function computeMedian(uint256[] data, uint256 size) internal pure returns (uint256) {
+        assert(size > 0 && data.length >= size);
+        for (uint256 i = 1; i < size; i++) {
+            for (uint256 j = i; j > 0 && data[j-1]  > data[j]; j--) {
+                uint256 tmp = data[j];
+                data[j] = data[j-1];
+                data[j-1] = tmp;
+            }
+        }
+        if (size % 2 == 1) {
+            return data[size / 2];
+        } else {
+            return data[size / 2].add(data[size / 2 - 1]) / 2;
+        }
+    }
+
+    function computeMedianQuickSort(uint256[] data, uint256 size) internal pure returns (uint256) {
+        quickSort(data, 0, size - 1);
+        if (size % 2 == 1) {
+            return data[size/2];
+        } else {
+            return data[size/2].add(data[size/2-1])/2;
+        }
+    }
+
+    function quickSort(uint256[] data, uint256 start, uint256 end) internal pure {
```

Comment:
Just question wrt. Recursive approach. 
Is there a potential chance we hit the call-stack limit?

Reply:
It needs few thousands for that to happen. I removed all the quick.* methods and only left the insertion sort version.